bot/exts/backend/sync/_syncers.py

```diff
@@ -183,12 +200,47 @@ async def _get_diff(self, guild: Guild) -> _Diff:
 
         return _Diff(users_to_create, users_to_update, None)
 
+    async def _get_users(self, endpoint: str = "bot/users", query_params: list = None) -> t.List[dict]:
+        """GET all users recursively."""
+        users = []
+        response: dict = await self.bot.api_client.get(endpoint, params=query_params)
+        users.extend(response["results"])
+
+        # The `response` is paginated, hence check if next page exists.
+        if (next_page_url := response["next"]) is not None:
+            next_endpoint, query_params = self.get_endpoint(next_page_url)
+            users.extend(await self._get_users(next_endpoint, query_params))
+        return users
+
+    @staticmethod
+    def get_endpoint(url: str) -> t.Tuple[str, t.List[tuple]]:
+        """Extract the API endpoint and query params from a URL."""
+        url = urlparse(url)
+
+        # Do not include starting `/` for endpoint.
+        endpoint = url.path[1:]
+
+        # Query params.
+        params = parse_qsl(url.query)
+
+        return endpoint, params
```

Comment:
Considering customising the pagination style on the API side to avoid having to parse URLs. An ideal style would be one that only returns page numbers rather than entire links.